src/lib/djerba/plugins/tar/Rscripts/convert_vep92_to_filtered_cbio.r

```diff
@@ -0,0 +1,148 @@
+ df_anno <- transform(df_anno,
+    oncogenic_binary = ifelse(ONCOGENIC == "Oncogenic" | ONCOGENIC == "Likely Oncogenic",
+                        "YES", "NO")
+ )
+
+ print("--- adding common_variant binary column ---")
+
+ # add common_variant yes or no columns
+ df_anno <- transform(df_anno,
+    ExAC_common = ifelse(grepl("common_variant", df_anno$FILTER),
+                        "YES", "NO")
+ )
+
+ print("--- adding gnomAD_AF_POPMAX binary column ---")
+
+ # add POPMAX yes or no columns
+ gnomad_cols <- c("gnomAD_AFR_AF", "gnomAD_AMR_AF", "gnomAD_ASJ_AF", "gnomAD_EAS_AF", "gnomAD_FIN_AF", "gnomAD_NFE_AF", "gnomAD_OTH_AF", "gnomAD_SAS_AF")
+ df_anno[gnomad_cols][is.na(df_anno[gnomad_cols])] <- 0
+ df_anno[, "gnomAD_AF_POPMAX"] <- apply(df_anno[gnomad_cols], 1, max)
+
+ print("--- small change to filters ---")
+ # caller artifact filters
+ df_anno$FILTER <- gsub("^clustered_events$",
+                        "PASS",
+                        df_anno$FILTER)
+
+ df_anno$FILTER <- gsub("^clustered_events;common_variant$",
+                        "PASS",
+                        df_anno$FILTER)
+
+ df_anno$FILTER <- gsub("^common_variant$",
+                        "PASS",
+                        df_anno$FILTER)
+
+ df_anno$FILTER <- gsub(".",
+                        "PASS",
+                        df_anno$FILTER,
+                        fixed=TRUE)
+
+ # Artifact Filter
+ print("--- artifact filter ---")
+ df_anno <- transform(df_anno,
+  TGL_FILTER_ARTIFACT = ifelse(FILTER == "PASS",
+                      "PASS", "Artifact")
+ )
+
+ # ExAC Filter
+ print("--- exac filter ---")
+ df_anno <- transform(df_anno,
+  TGL_FILTER_ExAC = ifelse(ExAC_common == "YES" & Matched_Norm_Sample_Barcode == "unmatched",
+                      "ExAC_common", "PASS")
+ )
+
+ # gnomAD_AF_POPMAX Filter
+ print("--- gnomAD filter ---")
+ df_anno <- transform(df_anno,
+  TGL_FILTER_gnomAD = ifelse(gnomAD_AF_POPMAX > 0.001 & Matched_Norm_Sample_Barcode == "unmatched",
+                      "gnomAD_common", "PASS")
+ )
+
+ # VAF Filter
+ print("--- VAF filter ---")
+ df_anno <- transform(df_anno,
+  TGL_FILTER_VAF = ifelse(tumour_vaf >= 0.15 | (tumour_vaf < 0.15 & oncogenic_binary == "YES"),
+                      "PASS", "low_VAF")
+ )
+
+ # Mark filters
+ print("--- printing verdict ---")
+ df_anno <- transform(df_anno,
+  TGL_FILTER_VERDICT = ifelse(TGL_FILTER_ARTIFACT == "PASS" & TGL_FILTER_ExAC == "PASS" & TGL_FILTER_gnomAD == "PASS" & TGL_FILTER_VAF == "PASS",
+                      "PASS",
+                       paste(df_anno$TGL_FILTER_ARTIFACT, df_anno$TGL_FILTER_ExAC, TGL_FILTER_gnomAD, df_anno$TGL_FILTER_VAF, sep=";"))
+ )
+
+ return(df_anno)
+ }
+
+construct_whizbam_links <- function(df, whizbam_url, studyid, tumourid, normalid, seqtype, genome) {
+   #whizbam_url <- "whizbam.oicr.on.ca"
+   #studyid <- "OCTCAP"
+   #tumourid <- "OCT_011408-TS"
+   #normalid <- "OCT-01-1408-BC-10"
+   #seqtype <- "GENOME"
+   #genome <- "hg38"
```

Comment:
test data should be in the testing scripts, not in the function

Reply:
it is also vep 105 ( and will continue to evolve) we can generalize this name if we like